### PR TITLE
adding product dmc-vet-search

### PR DESCRIPTION
**Pull request by jgarnhart-va**
This PR adds the product `dmc-vet-search` to the deployer.

The `dmc-vet-search` app is very much like `carma-search` in that it is able to search MVI  and other VA endpoints for veteran demographic and identity information.

### CODE REVIEW

**Comment by bschofield-va** [comment]
I'll merge this in after @mramirez-va or @sbair-va approve

File: products/dmc-vet-search.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  name: $DU_NAMESPACE
+  labels:
+    deployment-id: $K8S_DEPLOYMENT_ID
+    deployment-unit: $PRODUCT
+    deployment-unit-artifact: $DU_ARTIFACT
+    deployment-unit-version: $DU_VERSION
+    deployment-date: $BUILD_DATE
+    deployment-s3-bucket: $DU_AWS_BUCKET_NAME
+    deployment-s3-folder: $DU_S3_FOLDER
+    deployment-app-version: $CFC_VERSION
```
Comment (mramirez-va) [inline products/dmc-vet-search.yaml]: Isn't `CFC_VERSION` the CARMA fms connector version? Should this be `DMCS_VERSION` or something?

File: products/dmc-vet-search.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        paths:
+          - path: /dmc-vet-search/v0/(.*)
+            backend:
+              serviceName: dmc-vet-search
+              servicePort: 80
+---
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: carma-fms-resource-quota
```
Comment (mramirez-va) [inline products/dmc-vet-search.yaml]: This is will redeploy the CARMA fms resource quota...

Comment (jgarnhart-va) [reply products/dmc-vet-search.yaml]: Good catch. I will fix this.

File: products/dmc-vet-search.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  rules:
+    - http:
+        paths:
+          - path: /dmc-vet-search/v0/(.*)
+            backend:
+              serviceName: dmc-vet-search
+              servicePort: 80
+---
+apiVersion: v1
+kind: ResourceQuota
```
Comment (mramirez-va) [inline products/dmc-vet-search.yaml]: Setting limits at the namespace level without setting them at the pod level first will prevent this app from deploying at all. If the idea is to use the same limits as CARMA fms then take a look at how pod limits are set there for reference: https://github.com/department-of-veterans-affairs/health-apis-carma-fms-connector-deployment/blob/master/deployment.yaml#L25

File: products/dmc-vet-search.yaml (hunk 1 above)
Comment (jgarnhart-va) [reply products/dmc-vet-search.yaml]: Fixed.

File: products/dmc-vet-search.yaml (hunk 2 above)
Comment (jgarnhart-va) [reply products/dmc-vet-search.yaml]: Fixed

File: products/dmc-vet-search.yaml (hunk 3 above)
Comment (jgarnhart-va) [reply products/dmc-vet-search.yaml]: Set limits at the pod level.

https://github.com/department-of-veterans-affairs/health-apis-dmc-vet-search-deployment/blob/master/deployment.yaml

File: products/dmc-vet-search.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+metadata:
+  name: dmc-vet-search-resource-quota
+  namespace: $DU_NAMESPACE
+spec:
+  hard:
+    limits.cpu: "600m"
```
Comment (mramirez-va) [inline products/dmc-vet-search.yaml]: CPU will need to be bumped to `1800m` since there are 3 replicas and memory to `6G`

Comment (jgarnhart-va) [reply products/dmc-vet-search.yaml]: Fixed.

**Pull request merged**